.github/workflows/publish-docs.ym

```diff
@@ -0,0 +1,32 @@
+name: Publish Documentation
+
+on:
+  push:
+    tags:
+      - '*'
+  workflow_dispatch:
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v2
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14' 
+
+    - name: Install Dependencies
+      run: npm install
+
+    - name: Generate Documentation
+      run: npx typedoc
+
+    - name: Deploy to GitHub Pages
+      uses: JamesIves/github-pages-deploy-action@4.1.4
```

Comment:
Should use a pinned dependency

Reply:
Thank you, @frankhinek , for the suggestion to use a pinned dependency for Typedoc. I understand the importance of ensuring consistency and avoiding potential issues with automatic version updates. I have updated the GitHub Actions workflow file to use a specific version of Typedoc. This change will help maintain the stability of our documentation generation process. The updated version is now specified in the `.github/workflows/publish-docs.yml` file. Please let me know if there are any other improvements or changes needed.